taxcalc/functions.py

```diff
@@ -216,19 +216,20 @@ def EI_FICA(   e00900, e02100, SS_Income_c, e00200,
 
 @iterate_jit(parameters=["puf", "STD", "STD_Aged", "II_rt1", "II_rt2", "II_rt3", "II_rt4", 
              "II_rt5", "II_rt6", "II_rt7", "II_brk1", "II_brk2", "II_brk3", "II_brk4", "II_brk5", 
-            "II_brk6"], nopython=True, puf=True)
+            "II_brk6", "std2008"], nopython=True, puf=True)
 def StdDed( DSI, _earned, STD, e04470, e00100, e60000,
             MARS, MIDR, e15360, AGEP, AGES, PBI, SBI, _exact, e04200, e02400, STD_Aged,
             c04470, c00100, c21060, c21040, e37717, c04600, e04805, t04470, 
             f6251, _feided, c02700, FDED, II_rt1, II_rt2, II_rt3, II_rt4, II_rt5, II_rt6, II_rt7,
-            II_brk1, II_brk2, II_brk3, II_brk4, II_brk5, II_brk6, _fixup, puf):
+            II_brk1, II_brk2, II_brk3, II_brk4, II_brk5, II_brk6, _fixup, std2008, puf):
 
     if DSI == 1:
         c15100 = max(350 + _earned, STD[6])
     else:
         c15100 = 0.
 
-    if e04470 > 0 and e04470 < STD[MARS-1]:
+    # std2008a = [5450, 10900, 5450, 8000, 10900, 5450, 900]     
+    if FDED==1 and (e04470 < std2008[MARS-1]): #TODO should be 2008 vals, not current law.  
```

Comment:
Ok. I moved this stuff over to records.py. Two improvements for the future are probably to 

1. Keep imputations in a separate module
2. See if we can move this to vanilla python with a decorator.